src/libraries/System.Security.Cryptography/tests/X509Certificates/ExportTests.cs

```diff
@@ -441,44 +442,168 @@ public static void ECDsa_Export_DefaultKeyStorePermitsUnencryptedExports_ExportP
+                ECParameters actual = imported.ExportParameters(true);
+                ECParameters expected = ecdh.ExportParameters(true);
+
+                Assert.Equal(expected.D, actual.D);
+                Assert.Equal(expected.Q.X, actual.Q.X);
+                Assert.Equal(expected.Q.Y, actual.Q.Y);
             }
-            else if (typeof(TKey) == typeof(RSA))
+        }
+
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.iOS | TestPlatforms.MacCatalyst | TestPlatforms.tvOS, "The PKCS#12 Exportable flag is not supported on iOS/MacCatalyst/tvOS")]
+        public static void DSA_Export_DefaultKeyStorePermitsUnencryptedExports_ExportParameters()
+        {
+            (byte[] pkcs12, DSA dsa) = CreateSimplePkcs12<DSA>();
+
+            using (dsa)
             {
-                RSA rsaKey = RSA.Create(2048);
-                req = new("CN=simple", rsaKey, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
-                key = (TKey)(object)rsaKey;
+                using X509Certificate2 cert = new X509Certificate2(pkcs12, "", X509KeyStorageFlags.Exportable);
+                using DSA key = cert.GetDSAPrivateKey();
+                DSAParameters expected = dsa.ExportParameters(true);
+                DSAParameters actual = key.ExportParameters(true);
+
+                Assert.Equal(expected.X, actual.X);
             }
-            else
+        }
+
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.iOS | TestPlatforms.MacCatalyst | TestPlatforms.tvOS, "The PKCS#12 Exportable flag is not supported on iOS/MacCatalyst/tvOS")]
+        public static void DSA_Export_DefaultKeyStorePermitsUnencryptedExports_Pkcs8PrivateKey()
+        {
+            (byte[] pkcs12, DSA dsa) = CreateSimplePkcs12<DSA>();
+
+            using (dsa)
             {
-                throw new InvalidOperationException();
+                using X509Certificate2 cert = new X509Certificate2(pkcs12, "", X509KeyStorageFlags.Exportable);
+                using DSA key = cert.GetDSAPrivateKey();
+                byte[] exported = key.ExportPkcs8PrivateKey();
+
+                using DSA imported = DSA.Create();
+                imported.ImportPkcs8PrivateKey(exported, out _);
+                DSAParameters actual = imported.ExportParameters(true);
+                DSAParameters expected = dsa.ExportParameters(true);
+
+                Assert.Equal(expected.X, actual.X);
             }
+        }
+
+        private static (byte[] Pkcs12, TKey key) CreateSimplePkcs12<TKey>() where TKey : AsymmetricAlgorithm
+        {
+            using (ECDsa ca = ECDsa.Create(ECCurve.NamedCurves.nistP256))
+            {
+                CertificateRequest issuerRequest = new CertificateRequest(
+                    new X500DistinguishedName("CN=root"),
+                    ca,
+                    HashAlgorithmName.SHA256);
+
+                issuerRequest.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(true, false, 0, true));
```

Comment:
This could be

```suggestion
                    X509BasicConstraintsExtension.CreateForCertificateAuthority());
```

but that's just aesthetics.